src/regdesk/printer_manager.js

```diff
@@ -0,0 +1,171 @@
+import { LabelEpl, LP2844 } from 'WebZLP/src/LP2844'
+import { PrinterDropdown } from './printer_dropdown.js';
+
+export class PrinterManager {
+
+  #adultDropdown;
+  #minorDropdown;
+  #dropdowns;
+
+  #printerType = LP2844;
+
+  /**
+   * Initializes a new instance of the PrinterManager class.
+   *
+   * @param {Navigator} nav - Browser Navigator to hook up events to.
+   * @param {PrinterDropdown} adultDropdown - The dropdown object managing the adult printer.
+   * @param {PrinterDropdown} minorDropdown - The dropdown object managing the Minor printer.
+   */
+  constructor(nav, adultDropdown, minorDropdown) {
+    this.#adultDropdown = adultDropdown;
+    this.#adultDropdown.addEventListener(PrinterDropdown.ConnectEventName, async (e) => this.handleRequestConnect(e));
+    this.#adultDropdown.addEventListener(PrinterDropdown.DisconnectEventName, async (e) => this.handleRequestDisconnect(e));
+
+    this.#minorDropdown = minorDropdown;
+    this.#minorDropdown.addEventListener(PrinterDropdown.ConnectEventName, async (e) => this.handleRequestConnect(e));
+    this.#minorDropdown.addEventListener(PrinterDropdown.DisconnectEventName, async (e) => this.handleRequestDisconnect(e));
+
+    // Convenience array
+    this.#dropdowns = [this.#adultDropdown, this.#minorDropdown];
+
+    nav.usb.addEventListener('connect', async (e) => this.handleConnectPrinter(e));
```

Comment:
We don't do anything with this printer, worse off it gets garbage collected before we can disconnect, should we be handling this?

Reply:
The object lifetimes for USBDevices are interesting as it will keep handing you the same instance of the same object.

When you call, for example, navigator.usb.requestDevice while you're already using a device somewhere else you won't be handed a _new_ USBDevice object, instead it'll be the same instance of the one you're using elsewhere. Closing the 'new' one we're handed would close the connection out from the other printer.

At some point I'd like to make some utility wrapper or similar to PrinterManager in my library so it can better manage object lifecycles more directly because this surprised me, so it'll surprise others. I'm not able to find a great citation for this information, it was just handed to me by Some Guy I Know Who For Some Reason Knows A Lot About This Stuff ;)

Anyway, any situation in which we're handed a device object we should only dispose of it if we know we're the sole users of the object. This is fine for when we're handed a third printer when two are already hooked up, it's not owned by someone else and therefore we can dispose of it safely. In other cases though we aren't the sole owners and if anything we were handed the device by mistake and shouldn't do anything other than drop it on the floor as it's not our instance to mess with.